src/services/itemLogin/schemas.ts

```diff
@@ -1,83 +1,88 @@
-import { S } from 'fluent-json-schema';
+import { Type } from '@sinclair/typebox';
 
-import { ItemLoginSchemaStatus, ItemLoginSchemaType } from '@graasp/sdk';
-
-import { error, idParam, uuid } from '../../schemas/fluent-schema';
-import { item } from '../item/schema';
+import { FastifySchema } from 'fastify';
 
-export const credentials = S.object()
-  .additionalProperties(false)
-  .prop('username', S.string().minLength(3).maxLength(50).pattern('^\\S+( \\S+)*$'))
-  .prop('password', S.string().minLength(3).maxLength(50).pattern('^\\S+( \\S+)*$'))
-  .prop('memberId', uuid)
-  .oneOf([S.required(['username']), S.required(['memberId'])]);
+import { ItemLoginSchemaStatus, ItemLoginSchemaType } from '@graasp/sdk';
 
-const loginSchemaType = S.string().enum(Object.values(ItemLoginSchemaType));
-const loginSchemaStatus = S.string().enum(Object.values(ItemLoginSchemaStatus));
+import { customType, registerSchemaAsRef } from '../../plugins/typebox';
+import { entityIdSchemaRef, errorSchemaRef } from '../../schemas/global';
+import { accountSchemaRef } from '../account/schemas';
+import { itemSchemaRef } from '../item/schema';
 
-export const loginSchema = S.object()
-  .additionalProperties(false)
-  .prop('type', loginSchemaType)
-  .prop('status', loginSchemaStatus)
-  .prop('item', item)
-  .prop('createdAt', S.string())
-  .prop('updatedAt', S.string())
-  .prop('id', uuid);
+export const credentialsSchemaRef = registerSchemaAsRef(
+  'itemLoginCredentials',
+  'Item Login Credentials',
+  Type.Object(
+    {
+      username: Type.String({ minLength: 3, maxLength: 50, pattern: '^\\S+( \\S+)*$' }),
```

Comment:
I don't know if we want to extract the pattern in the same const for the username and password for allowing their to evolve independently, but should we extract the patterns in varioble or other, to simplify the readability of the Regex?

Reply:
It'll use the customType username in a future PR 👍🏻